### DISCUSSION TITLE
Modified request headers appear on flow but not on client

### DISCUSSION THREAD

**Discussion by devcatkit** [mitmproxy/mitmproxy#5735] [General]
hi. im new noobie here. trying to set up reverse proxy for a project and wanting to modify request headers to change cookie. 

when using command in terminal --modify-headers everything works fine, but when writing a python script everything goes wrong

`from mitmproxy import http`

`def request(flow: http.HTTPFlow) -> None:`
   `flow.request.headers["cookie"] = "mycookie"`

using this script I can see that the cookie is added in flow (in terminal) but when looking at network requests from browser the cookie is not there and no changes are made. Some help is really more than appreciate. 

thank you all.

**Comment by mhils** [comment]
You are modifying a request cookie while it is in transit from mitmproxy to the target server. There's no mechanism for the client to learn that the headers have been fiddled with after they sent them.

Likewise, if you modify response headers, those changes will only appear on the client; the server has now way to notice the changes being done in transit.

**Reply by devcatkit** [reply]
hello. first i want to say really thanks for your help. 

i figured this just now. i manage to get the response on the client side. basically for some reason set-cookie not work so i am manually trying to get the set-cookie and set it as the cookie because otherwise server drops conection. 

so far i got this code 

`from mitmproxy import http`


`class ModifyHeader:`

  ` def response(self, flow: http.HTTPFlow):`
       `set_cookie_str = flow.response.headers.get("set-cookie", "")`
       ` print("set_cookie", set_cookie_str)`

`addons = [`
   `ModifyHeader()`
`]`

from what i can tell, somehow i need to refresh and set that cookie using flow.response.header but not sure exactly how yet. 

thanks very much
